### PR TITLE
Update ConfigurationManagerTest to write to temp directory

### LINKED ISSUE #17199: Some S.Configuration tests are writing to the test directory
Tests should not write to the current directory. They should use `System.IO.TempDirectory/.TempFile` or similar means to write to a unique location in temp. This causes failures in appcontainer as well as being messy.

Similar issue: https://github.com/dotnet/corefx/issues/17072

```
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output: ERROR:    MonoTests.System.Configuration.ConfigurationManagerTest.TestContext2 [FAIL]
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:       System.Configuration.ConfigurationErrorsException : An error occurred loading a configuration file: Access to the path 'C:\dotnetbuild\work\739d31a0-6c27-4764-813e-e3d8e36bec8f\Work\f09fd077-868e-48d9-8a60-5ecad40cb3db\Unzip\e1wis4c1.tmp' is denied. (C:\dotnetbuild\work\739d31a0-6c27-4764-813e-e3d8e36bec8f\Work\f09fd077-868e-48d9-8a60-5ecad40cb3db\Unzip\0cv2teiu.1jd.config)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:       ---- System.UnauthorizedAccessException : Access to the path 'C:\dotnetbuild\work\739d31a0-6c27-4764-813e-e3d8e36bec8f\Work\f09fd077-868e-48d9-8a60-5ecad40cb3db\Unzip\e1wis4c1.tmp' is denied.
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:       Stack Trace:
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.Configuration.SaveAsImpl(String filename, ConfigurationSaveMode saveMode, Boolean forceSaveAll)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.Configuration.Save()
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at MonoTests.System.Configuration.ConfigurationManagerTest.TestContext2()
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:         ----- Inner Stack Trace -----
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.IO.FileStream.OpenHandle(FileMode mode, FileShare share, FileOptions options)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access, FileShare share, Int32 bufferSize, FileOptions options)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.IO.FileStream..ctor(String path, FileMode mode, FileAccess access)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.IO.Internal.TempFileCollection.EnsureTempNameCreated()
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.IO.Internal.TempFileCollection.get_BasePath()
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.IO.Internal.TempFileCollection.AddExtension(String fileExtension, Boolean keepFile)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.IO.Internal.TempFileCollection.AddExtension(String fileExtension)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.Internal.WriteFileContext..ctor(String filename, String templateFilename)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.Internal.InternalConfigHost.StaticOpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.Internal.InternalConfigHost.System.Configuration.Internal.IInternalConfigHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.ClientConfigurationHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.UpdateConfigHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.Internal.DelegatingConfigHost.OpenStreamForWrite(String streamName, String templateStreamName, Object& writeContext)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:            at System.Configuration.MgmtConfigurationRecord.SaveAs(String filename, ConfigurationSaveMode saveMode, Boolean forceUpdateAll)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output: ERROR:    MonoTests.System.Configuration.ConfigurationManagerTest.TestFileMap [FAIL]
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:       System.Configuration.ConfigurationErrorsException : An error occurred loading a configuration file: Access to the path 'C:\dotnetbuild\work\739d31a0-6c27-4764-813e-e3d8e36bec8f\Work\f09fd077-868e-48d9-8a60-5ecad40cb3db\Unzip\tagheare.tmp' is denied. (C:\dotnetbuild\work\739d31a0-6c27-4764-813e-e3d8e36bec8f\Work\f09fd077-868e-48d9-8a60-5ecad40cb3db\Unzip\amsxwqqf.crl.config)
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:       ---- System.UnauthorizedAccessException : Access to the path 'C:\dotnetbuild\work\739d31a0-6c27-4764-813e-e3d8e36bec8f\Work\f09fd077-868e-48d9-8a60-5ecad40cb3db\Unzip\tagheare.tmp' is denied.
2017-03-16 04:45:27,631: INFO: proc(54): run_and_log_output: Output:       Stack Trace:

```

### PR DESCRIPTION
**Pull request by StanislavUshakov**
Use TempDirectory instead of the current directory where the test is running

Fix #17199

### CODE REVIEW

File: src/System.Configuration.ConfigurationManager/tests/Mono/ConfigurationManagerTest.cs [hunk 1]
```diff
@@ -320,13 +320,13 @@ public void TestAddRemoveSection()
         [Fact]
         public void TestFileMap()
         {
-            var name = Path.GetRandomFileName() + ".config";
-            Assert.False(File.Exists(name));
-
-            try
+            using (var temp = new TempDirectory())
             {
+                var configPath = Path.Combine(temp.Path, Path.GetRandomFileName() + ".config");
```
Comment (stephentoub) [inline src/System.Configuration.ConfigurationManager/tests/Mono/ConfigurationManagerTest.cs]: Nit: var => string

File: src/System.Configuration.ConfigurationManager/tests/Mono/ConfigurationManagerTest.cs [hunk 2]
```diff
@@ -373,13 +369,13 @@ public void TestContext()
         [Fact]
         public void TestContext2()
         {
-            var name = Path.GetRandomFileName() + ".config";
-            Assert.False(File.Exists(name));
-
-            try
+            using (var temp = new TempDirectory())
             {
+                var configPath = Path.Combine(temp.Path, Path.GetRandomFileName() + ".config");
```
Comment (stephentoub) [inline src/System.Configuration.ConfigurationManager/tests/Mono/ConfigurationManagerTest.cs]: Nit: var => string

**Comment by danmosemsft** [comment]
@StanislavUshakov this is a similar issue if you're interested: https://github.com/dotnet/corefx/issues/17072

**Comment by JeremyKuhne** [comment]
Thanks for cleaning this one up!

Note that I tried to leave the tests from Mono as close to the original source as possible. We should absolutely fix test issues in these, just try not to refactor them too much. (Which you haven't.)

**Comment by StanislavUshakov** [comment]
Thank for quick review.
Yes, I'd like to take  #17072

**Pull request merged**